Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -66,38 +42,76 @@ import BraintreeCore
+    ///    - additionalInformation: Optional. The additional information used for verification.
+    ///    - billingAddress: Optional. The billing address used for verification
+    ///    - cardAddChallengeRequested: Optional.  An authentication created using this flag should only be used for vaulting operations (creation of customers' credit cards or payment methods) and not for creating transactions. If set to `true`, a card-add challenge will be requested from the issuer. If set to `false`, a card-add challenge will not be requested. If the parameter is missing, a card-add challenge will only be requested for $0 amount.
+    ///    - challengeRequested: Optional. If set to true, an authentication challenge will be forced if possible.
+    ///    - customFields: Optional. Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+    ///    - dataOnlyRequested:  Optional. Indicates whether to use the data only flow. In this flow, frictionless 3DS is ensured for Mastercard cardholders as the card scheme provides a risk score for the issuer to determine whether to approve. If data only is not supported by the processor, a validation error will be raised. Non-Mastercard cardholders will fallback to a normal 3DS flow.
+    ///    - dfReferenceID: Optional. The dfReferenceID for the session, particularly useful for merchants performing 3DS lookup.
+    ///    - email: Optional. The email used for verification.
+    ///    - exemptionRequested: Optional. If set to true, an exemption to the authentication challenge will be requested.
+    ///    - mobilePhoneNumber: Optional. The mobile phone number used for verification. Only numbers. Remove dashes, parentheses and other characters.
+    ///    - renderTypes: Optional: List of all the render types that the device supports for displaying specific challenge user interfaces within the 3D Secure challenge. When using `BTThreeDSecureUIType.both` or `BTThreeDSecureUIType.html`, all `BTThreeDSecureRenderType` options must be set. When using `BTThreeDSecureUIType.native`, all `BTThreeDSecureRenderType` options except `.html` must be set.
+    ///    - requestedExemptionType: Optional. The exemption type to be requested. If an exemption is requested and the exemption's conditions are satisfied, then it will be applied.
+    ///    - shippingMethod: Optional. The shipping method chosen for the transaction
+    ///    - uiType: Optional: Sets all UI types that the device supports for displaying specific challenge user interfaces in the 3D Secure challenge. Defaults to `.both`
+    ///    - v2UICustomization: Optional. UI Customization for 3DS2 challenge views.
+    public init(
+        amount: String,
```

Comment:
🤔  I am getting an error with the amount as a string in the demo app. Are we sure this is valid? It looks like a `Double` works as expected.

Reply:
hmm yea i'm a bit confused on this - i also feel amount as `Double` makes more sense (and this aligns with what is mentioned in our iOS BT Dev Docs) but after asking here  #[bt-service-3dsecure](https://paypal.slack.com/archives/C02743S2L/p1731618309941569) and checking the android [equivalent](https://github.com/braintree/braintree_android/blob/c9d4246eccb6926a0985b5e6d409147d6fb38470/ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecureRequest.kt#L56) it looks like they define amount as a string

Reply:
Ah, you're right, I must have been doing something weird the first time I tested it as a string on main that resulted in an error. If we want to follow what android/the service channel recommends that's okay with me.